src/functional/iterator/ir.py

```diff
@@ -1,11 +1,18 @@
 from typing import List, Union
 
-from eve import Node
+import eve
 from eve.traits import SymbolName, SymbolTableTrait
 from eve.type_definitions import SymbolRef
 from functional.iterator.util.sym_validation import validate_symbol_refs
 
 
+class Node(eve.Node):
```

Comment:
Would it make sense to make this class non-instantiable (`eve.utils.non_instantiable`)?

Reply:
Done.